delira/training/callbacks/abstract_callback.py

```diff
@@ -59,3 +59,41 @@ def at_epoch_end(self, trainer, **kwargs):
 
         """
         return {}
+
+    def at_training_begin(self, trainer, **kwargs):
+        """
+        Function which will be executed at begin of training
+
+        Parameters
+        ----------
+        trainer : :class:`AbstractNetworkTrainer`
+        **kwargs :
+            additional keyword arguments
+
+        Returns
+        -------
+        dict
+            modified trainer attributes, where the name must correspond to the
+            trainer's attribute name
+
+        """
+        return {}
+
+    def at_training_end(self, trainer, **kwargs):
```

Comment:
I think we should explicitly add the arguments.

Reply:
Same as above

Reply:
Closing this for the discussion in the other PR.